### ISSUE TITLE
error output hard-coded to GET verb adds confusion 

### ISSUE THREAD

**Issue by travi** [traverson/traverson#97]
I ran into a situation today that confused me enough to send me down the wrong path. I have a traversal that ends by doing a `GET` for a resource in order to get access to the links for actions against that resource. It then makes a `POST` request against one of those links that points to the same url. 

The result of trying to follow that traversal resulted in the following error:
```
170615/210440.374, [error] message: HTTP GET for <api url for the resource> resulted in HTTP status code 401.
```

This made me think that the `GET` request for the resource was failing, but it was actually the `POST` request that happened after the `GET` request to the same url.

It looks like the error message hard-codes the verb into the message:
https://github.com/traverson/traverson/blob/9f4b98ce8e280f4c2f99516b6e3f9ef0e953435a/lib/transforms/check_http_status.js#L43

Could this be made dynamic so that the correct verb is printed with the error?

**Comment by basti1302** [comment]
Are you sure? As far as I can tell, an HTTP error at the end of your traversal should just be handed back to you as is, traverson only checks the HTTP status of intermediate requests, and these are always GET requests. Can you provide a reproducible example?

**Comment by travi** [comment]
Yeah, the logs clearly showed that the error was for the `POST` request made by the `.post()` at the end of the chain. Fixing the `POST` call prevented the error from happening and removing that fix brings it back.

Unfortunately, the project and api that produced this issue are both private, so it would take me some time to get a reproducible example together. Willing to do so if needed, but I probably won't have a chance to do so right away.

In case it is helpful, here are the high level steps taken in this particular traversal:
```js
    apiRoot()
      .withTemplateParameters(pathToFollowValues)
      .follow(pathWithAction)
      .withRequestOptions(getAllRequestOptions({authToken, payload}, pathWithAction))
      .convertResponseToObject()
      .post(payload, (err, response) => handlePartialUpdateResponse(response, err, resolve, reject));
```

Looking a little deeper into the `traverson` source, my best guess is that it is related to the use of `.convertResponseToObject()`, which [applies additional processing](https://github.com/traverson/traverson/blob/d309b288003bb9c678795793efde55996c2362ff/lib/actions.js#L115-L122) to the `.post()` response.

**Comment by basti1302** [comment]
a fix has landed in 6.1.1
